src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,222 @@
+                } else {
+                    val maybeArchive = it.selectFirst("""td[colspan="3"] > a""")?.attr("href")
+                    if (maybeArchive != null) {
+                        mangas.add(dndManga(maybeArchive, nextMangaTitle, SManga.COMPLETED, nthManga))
+                        nthManga += 1
+                    } else {
+                        // We reached the end, assuming the page layout stays consistent beyond D&D8.
+                        // Thus, we append our final manga with this current page as its archive.
+                        // Unfortunately this means we will needlessly fetch this page twice.
+                        mangas.add(dndManga("/archive.html", nextMangaTitle, SManga.ONGOING, nthManga))
+                        return@stop
+                    }
+                }
+            }
+        }
+
+        return Observable.just(MangasPage(mangas, false))
+    }
+
+    // Not efficient, but the simplest way for me to refresh
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        fetchPopularManga(0)
+            .map { mangasPage ->
+                mangasPage
+                    .mangas
+                    .first { it.url == manga.url }!!
+            }
+
+    // This implementation here is needlessly complicated, for it has to automatically detect
+    // whether we’re in a date-annotated archive, the main archive, or a dateless archive.
+    // All three are largely similar, there are just *some* (annoying) differences we have to
+    // deal with.
+    private fun fetchChaptersForDatedAndDatelessArchives(archivePages: Document, archivePath: String): List<SChapter> {
+        val chapters = mutableListOf<SChapter>()
+        var i = 0
+        archivePages.select("""div.text > table.text > tbody > tr""").forEach {
+            val pageData = it.select("""td""")
+            var pageAnchor = pageData.getOrNull(2)?.selectFirst("a")
+            // null for »Intermission«, main archive, dateless archive,…
+            if (pageAnchor != null) {
+                val a = pageAnchor
+
+                chapters.add(
+                    SChapter.create().apply {
+                        name = a.text()
+                        chapter_number = i.toFloat()
+                        date_upload = DATE_FMT.parse(pageData[0].text().trim())?.time ?: 0L
+                        url = absolutePageUrl(a.attr("href"), archivePath)
+                    },
+                )
+                i += 1
+            } else if (!pageData.hasAttr("colspan")) {
+                // Are we in a dateless archive?
+                pageAnchor = pageData.getOrNull(0)?.selectFirst("a")
+                if (pageAnchor != null) {
+                    // For now we assume all dateless archives were published the same day
+                    // the Solo book was published, because currently Solo’s the only one such book.
+                    val pageDate = 1660435200000L + (i * 1000 * 60)
```

Comment:
What is the date? And is it better than the default behavior (date set to current time)?

Reply:
If you know the entry that the date applies to you should hardcode that, and basically what you wrote earlier. Assuming it may occur later it's better to default to 0 / 'today'

Reply:
In the next commit that date is sourced from the site footer, which is the date the corresponding page was published.